### DISCUSSION TITLE
link error while calling ImGui functions

### DISCUSSION THREAD

**Discussion by Dnurr** [ocornut/imgui#6455] [New Users Build/Link/Run issues ONLY!]
Hi,
I'm quite new with Dear ImGui...
When I build my project, the Visual Studio 2022 C++ compiler doesn't throw any errors, but I get it when I start a debug session.

Here I post the link errors that I get:
```
1>main.obj : error LNK2019: riferimento al simbolo esterno "struct ImGuiContext * __cdecl ImGui::CreateContext(struct ImFontAtlas *)" (?CreateContext@ImGui@@YAPEAUImGuiContext@@PEAUImFontAtlas@@@Z) non risolto nella funzione main
1>main.obj : error LNK2019: riferimento al simbolo esterno "void __cdecl ImGui::NewFrame(void)" (?NewFrame@ImGui@@YAXXZ) non risolto nella funzione main
1>main.obj : error LNK2019: riferimento al simbolo esterno "bool __cdecl ImGui::Begin(char const *,bool *,int)" (?Begin@ImGui@@YA_NPEBDPEA_NH@Z) non risolto nella funzione main
1>main.obj : error LNK2019: riferimento al simbolo esterno "void __cdecl ImGui::End(void)" (?End@ImGui@@YAXXZ) non risolto nella funzione main
1>main.obj : error LNK2019: riferimento al simbolo esterno "void __cdecl ImGui::Text(char const *,...)" (?Text@ImGui@@YAXPEBDZZ) non risolto nella funzione main
1>main.obj : error LNK2019: riferimento al simbolo esterno "bool __cdecl ImGui::DebugCheckVersionAndDataLayout(char const *,unsigned __int64,unsigned __int64,unsigned __int64,unsigned __int64,unsigned __int64,unsigned __int64)" (?DebugCheckVersionAndDataLayout@ImGui@@YA_NPEBD_K11111@Z) non risolto nella funzione main
```

When I try to call ImGui functions in my main (or other) function (I know this isn't the entire code it's only an example for throwing link errors and show you it):
```
int main()
{
	IMGUI_CHECKVERSION();
	CreateContext();
	NewFrame();
	Begin("Engine");
	Text("test");
	End();
	
	return 0;
}
```

I set my main project's settings to communicate with imgui folder (where could be found all .h and .cpp files)

Lastly, I built ImGui .cpp files into a .lib file with CMake and set it in Additional Dependecies.

Thank you in advance.

---------------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------------------

EDIT: I had to add the entire path of .lib file.

**Comment by ocornut** [comment]
> EDIT: I had to add the entire path of .lib file.

This error has been happening over and over for 40+ years for C/C++ developers and the answer is the same in 90% of cases.
If you get a link error it means you are not linking your app with the compiled files you aim to use.
